src/artifacts.js

```diff
@@ -175,8 +217,46 @@ api.declare({
   // Construct details for different storage types
   var isPublic = /^public\//.test(name);
   var details  = {};
+  let uploadId;
   switch (storageType) {
+    case 'blob':
+      // Generate the details that we'll be using.
+      details.size = input.size;
+      details.sha256 = input.sha256;
+      if (input.parts) {
+        details.parts = input.parts;
+      }
+      // TODO: These ought to be configurable
+      details.provider = 's3'; // MAKECONFIG
+      details.region = 'us-east-1'; // MAKECONFIG
+      if (input.encoding) {
+        details.encoding = input.encoding;
+      }
+
+      if (isPublic) {
+        details.bucket = 'taskcluster-blobs'; // MAKECONFIG
+      } else {
+        details.bucket = 'taskcluster-private-blobs'; // MAKECONFIG
+      }
+
+      details.key = encodeBlobKey(taskId, runId, name);
```

Comment:
whether we do the MD5 thing or not, I still think that having a single place where we map taskId, runId and name to an S3 key is worthwhile.